tests/typ/compiler/array.typ

```diff
@@ -390,3 +390,16 @@
 // Keyed pair after this is already identified as an array.
 // Error: 6-14 expected expression, found keyed pair
 #(1, "key": 2)
+
+---
```

Comment:
Due to https://github.com/typst/typst/pull/3922, this will need to be moved to `tests/suite/foundations/array.typ` and the individual tests need to be named.